### PR TITLE
feat: add a new plugin stack_overflow

### PR DESCRIPTION
**Pull request by LinkW77**
# Pull Request

## PR Description

Add a new plugin Stack Overflow to solve programming questions

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Performance enhancement
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other, please describe:

## Checklist

Before submitting this PR, please make sure:

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guidelines.
- [x] I have tested my changes locally to ensure they are effective.
- [x] I have updated the necessary documentation (if applicable).

### CODE REVIEW

File: plugin/bundles/stack_overflow/resources/i18n/en.yml
Comment (DynamesC) [inline plugin/bundles/stack_overflow/resources/i18n/en.yml:1]: rename input 'not_tagged''s name to human reading format, for example 'Not Tagged'. And also update its description to explain more

File: plugin/bundles/stack_overflow/resources/bundle_schema.yml
Comment (DynamesC) [inline plugin/bundles/stack_overflow/resources/bundle_schema.yml:1]: provider should be: StackOverflow

**Review by taskingaijc — APPROVED** [review]
LGTM

**Review by DynamesC — APPROVED** [review]
LGTM

File: plugin/bundles/stack_overflow/plugins/stack_overflow_search/plugin.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class StackOverflowSearch(PluginHandler):
+    async def execute(self, credentials: BundleCredentials, plugin_input: PluginInput) -> PluginOutput:
+        tagged: str = plugin_input.input_params.get("tagged", "")
+        not_tagged: str = plugin_input.input_params.get("not_tagged", "")
+        title: str = plugin_input.input_params.get("title", "")
+        sort: str = plugin_input.input_params.get("sort", "")
+        order: str = plugin_input.input_params.get("order", "")
+
+        url = f"https://api.stackexchange.com/2.3/search?site=stackoverflow&tagged={tagged}&nottagged={not_tagged}&intitle={title}&sort={sort}&order={order}"
```
Comment (DynamesC) [inline plugin/bundles/stack_overflow/plugins/stack_overflow_search/plugin.py:17]: Only append parameters when the parameter is presented

**Review by DynamesC — CHANGES_REQUESTED** [review]

**Review by DynamesC — APPROVED** [review]
LGTM

**Review by taskingaijc — APPROVED** [review]
LGTM

**Pull request merged**